### PR TITLE
Improve development environment networking

### LINKED ISSUE #76: Don't require changes to hosts file to run a basic development deployment
Specifically, accessing a development server by its IP address (`http://172.28.128.100/`) should not cause a 301 redirect to `http://isic-archive.dev`, which necessitates that `/etc/hosts` be updated. However, this behavior (redirecting all other host requests to the canonical host) is desirable in production, and should be preserved there.

Perhaps two different Nginx files (for development and production) should be made, rather than adding lots of extra template conditionals to the existing `girder.conf.j2`.

### LINKED ISSUE #82: Vagrant private network fails on Windows
Running `vagrant up` on Windows gives an error when setting up the private network (i.e. host-only networking in VirtualBox):

```
There was an error while executing `VBoxManage`, a CLI used by Vagrant
for controlling VirtualBox. The command and stderr is shown below.

Command: ["hostonlyif", "create"]

Stderr: 0%...
Progress state: E_INVALIDARG
VBoxManage.exe: error: Failed to create the host-only adapter
VBoxManage.exe: error: Assertion failed: [!aInterfaceName.isEmpty()] at 'F:\tinderbox\win-rel\src\VBox\Main\src-server\HostNetworkInterfaceImpl.cpp' (74) in long __cdecl HostNetworkInterface::init(class com::Bstr,class com::Bstr,class com::Guid,enum __MIDL___MIDL_itf_VirtualBox_0000_0000_0038).
VBoxManage.exe: error: Please contact the product vendor!
VBoxManage.exe: error: Details: code E_FAIL (0x80004005), component HostNetworkInterfaceWrap, interface IHostNetworkInterface
VBoxManage.exe: error: Context: "enum RTEXITCODE __cdecl handleCreate(struct HandlerArg *)" at line 71 of file VBoxManageHostonly.cpp
```

Even though it's possible to work around this error by a series of manual steps, accessing the VM from the host at port 80 never works.

A workaround is to:
- disable the private network in the Vagrantfile
- port forward localhost:8080 to VM:80
- Add the following to `%SystemRoot%\System32\drivers\etc\hosts`:

> 127.0.0.1 isic-archive.dev
> 127.0.0.1 isic-archive.upstream

Then you can access isic-archive.dev and isic-archive.upstream at port 8080.

### PR DESCRIPTION
**Pull request by brianhelba**
Fixes #76 
Fixes #82

### CODE REVIEW

**Comment by brianhelba** [comment]
@msmolens PTAL at the code. Travis seems to be broken (their fault, I believe) right now, but I'll restart the tests when it's fixed.

Let's talk before you try to checkout and deploy this, as there's still an unfixed bug with the large_image plugin that will cause a fresh install to not fully work.

File: readme.rst [hunk 1]
```diff
@@ -25,84 +25,90 @@ Prerequisites
...
+  development machine.
 
-* Register a new user (this will be an admin user)
+* Register a new user (this will be an admin user).
 
 * Go to ``Admin console`` -> ``Plugins`` and enable the ``ISIC Archive`` plugin
   (and optionally the ``OAuth2 login`` plugin), then use the button at the top
   of the page to restart the server.
 
+* Visit ``http://127.0.0.1:8080/girder``in your web browser, as this is where
```
Comment (msmolens) [inline readme.rst]: Need to add a space after the URL.

File: readme.rst [hunk 2]
```diff
@@ -25,84 +25,90 @@ Prerequisites
...
+  ::
 
-  # from within your preferred source development directory
-  git clone https://github.com/ImageMarkup/isic-archive.git
-  cd isic-archive
-  git submodule init && git submodule update
+    # from within your preferred source development directory
+    git clone https://github.com/ImageMarkup/isic-archive.git
+    cd isic-archive
+    git submodule init && git submodule update
```
Comment (msmolens) [inline readme.rst]: If we want, this can be shortened to the equivalent `git submodule update --init`.

File: server/__init__.py [hunk 3]
```diff
@@ -102,6 +106,33 @@ def onDescribeResource(event):
...
+        # Job.save is overridden, the local model may be desynchronized from
+        # the database after saving; this is fine, since girder_worker
+        # (where it matters) will always load directly from the correct entry
+        # in the database
+        def replaceHost(url):
+            # TODO: get this from the server config or the request
+            return 'http://127.0.0.1:8080' + url[url.find('/api/v1'):]
+
+        job['kwargs'] = json_util.loads(job['kwargs'])
+        for specValue in itertools.chain(
```
Comment (msmolens) [inline server/__init__.py]: A small example of the expected structure of a job could make this code faster to understand.

**Comment by msmolens** [comment]
@brianhelba Great! I deployed this on a test machine, patched large_image as I have on my main machine, and ran some tests. Most everything works as expected, including thumbnail generation and using the Swagger interface. The only issue I saw is that the task dashboard redirects, such as when clicking to reveiw a dataset, lose the `:8080` port.

**Comment by brianhelba** [comment]
@msmolens I've fixed the 3 specific issues you noted, and rebased.

I'm still looking into the task redirect problem.

**Comment by brianhelba** [comment]
@msmolens I fixed the task redirect issue too, PTAL.

As an aside, I'd like to fix the large_image issue too, but that plugin has made a major change in their dependencies that I need to incorporate before we can update to a version that includes the bugfix. I'll try to work on that task soon.

**Comment by brianhelba** [comment]
Merging.

**Pull request merged**